### PR TITLE
`azapi_resource_action` - `locks`

### PR DESCRIPTION
**Pull request by ms-henglu**
Fixes https://github.com/Azure/terraform-provider-azapi/issues/178

### CODE REVIEW

File: docs/resources/azapi_resource_action.md [hunk 1]
```diff
@@ -84,6 +84,8 @@ The following arguments are supported:
 
 * `body` - (Optional) A JSON object that contains the request body.
 
+* `locks` - (Optional) A list of ARM resource IDs which are used to avoid modify azapi resources at the same time.
```
Comment (magodo) [inline docs/resources/azapi_resource_action.md:87]: This looks like not necessarily need to be a resource ID?

File: internal/services/azapi_resource_action_resource.go [hunk 2]
```diff
@@ -121,6 +131,12 @@ func resourceResourceActionCreateUpdate(d *schema.ResourceData, meta interface{}
 	}
 
 	log.Printf("[INFO] request body: %v\n", body)
+
+	for _, id := range d.Get("locks").([]interface{}) {
```
Comment (magodo) [inline internal/services/azapi_resource_action_resource.go:135]: Should we also support it for delete?

**Review by magodo — CHANGES_REQUESTED** [review]

File: docs/resources/azapi_resource_action.md (hunk 1 above)
Comment (ms-henglu) [reply docs/resources/azapi_resource_action.md:87]: Yes, but I want to align this description with `azapi_resource` and users might get confused if it says any strings are acceptable.

File: internal/services/azapi_resource_action_resource.go (hunk 2 above)
Comment (magodo) [reply internal/services/azapi_resource_action_resource.go:135]: Oh, missed it was for action...

**Review by magodo — APPROVED** [review]

**Pull request merged**